### DISCUSSION TITLE
Private dependancies

### DISCUSSION THREAD

**Discussion by tkram01** [python-poetry/poetry#3800] [Q&A]
If I build a library with poetry which has dependancies on private pypi repos can the library be installed without adding the private repos to the toml? Or do you have to do the equivalent of setting --extra-index for all the dependancies in the top level project?

**Comment by abn** [comment]
No poetry does not support pulling packages from sources configured solely at runtime (ie. not in `pyproject.toml`). This is because, when a package is retrieved, poetry locks it with the metadata associated with it, including the source information.

If your package is not available in a public index, any other user trying to install the project, will have issues. Allowing something like this will cause more inconsistencies and problems than what it may solve.
